### DISCUSSION TITLE
On Android 12 - Note 20 (non rooted)- how to get USB WIFI device to detect?

### DISCUSSION THREAD

**Discussion by WilsonBradley** [termux/termux-app#3112] [General]
On Android 12 - Note 20 (non rooted)- how to get USB WIFI device to detect?
I have Termux running Nethunter container , however can not get a usb (rtl chipset) to detect when plugging in directly to phone..
Considering all the videos' , assumed this would work without needing kernel replacement on phone..

I did find this, but just sits there and never returns results;
https://wiki.termux.com/wiki/Termux-usb
termux-usb -l 

.

**Comment by sylirre** [comment]
This will not work.

`termux-usb` is supposed to be used with standalone USB devices, i.e. ones that just work after powered on and provide I/O stream and do not need complex configuration. This is: ~~USB flash drives~~ (only direct I/O, no file system mounting), USB serial line adapters, smart cards and maybe something else. USB Wi-Fi stick is not a standalone device because it needs firmware to be loaded and network interface on the host configured.

You don't have root. Considering this I can also assume you didn't compiled and flashed a custom kernel with drivers for your USB Wi-Fi stick, not saying that you also will need a firmware for it besides the drivers, which will require to modify /system partition (not possible without root). So, what we have:

* Your device can't see the USB Wi-Fi stick because of no drivers and firmware.
* You can't manage the network interface of the USB Wi-Fi stick because of absence of root permissions.

> Considering all the videos' , assumed this would work without needing kernel replacement on phone..

I will not judge whether videos you have watched aren't just fake/clickbait... but I assume you need to learn a lot. Begin with Bash and standard utilities, this will make you able to troubleshoot most of errors you are facing. Next your step is to root your device, build and install custom kernel and you'll definitely get some progress on your issue.

Pages at https://www.kali.org/docs/nethunter/ (including links) and especially https://wiki.termux.com/wiki/Hacking (!!!) also needs to be fully learned.

Remember that we do not provide help for installation, setting up, using and troubleshooting the hacking/phishing tools of whatever kind. That's really annoying how many people come to us after these "tutorials" and know nothing but want everything (hack something/someone, free quick cash, etc)...
